From plasmaroo@gentoo.org  Wed Aug 17 13:19:31 2005
Date: Wed, 17 Aug 2005 21:19:24 +0100
Subject: [PATCH] Revert unnecessary zlib_inflate/inftrees.c fix

From: Linus Torvalds <torvalds@osdl.org>

It turns out that empty distance code tables are not an error, and that
a compressed block with only literals can validly have an empty table
and should not be flagged as a data error.

Some old versions of gzip had problems with this case, but it does not
affect the zlib code in the kernel.

Analysis and explanations thanks to Sergey Vlasov <vsu@altlinux.ru>

Cc: Sergey Vlasov <vsu@altlinux.ru>
Cc: Tavis Ormandy <taviso@gentoo.org>
Cc: Tim Yamin <plasmaroo@gentoo.org>
Signed-off-by: Linus Torvalds <torvalds@osdl.org>
Signed-off-by: Chris Wright <chrisw@osdl.org>
---
 lib/zlib_inflate/inftrees.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.12.y/lib/zlib_inflate/inftrees.c
===================================================================
--- linux-2.6.12.y.orig/lib/zlib_inflate/inftrees.c
+++ linux-2.6.12.y/lib/zlib_inflate/inftrees.c
@@ -141,7 +141,7 @@ static int huft_build(
   {
     *t = NULL;
     *m = 0;
-    return Z_DATA_ERROR;
+    return Z_OK;
   }
 
 
